dnd5esheets/front/src/store/index.ts

```diff
@@ -1,491 +1,358 @@
-      8 +
-      scoreToSkillModifier(
-        character.data.scores[
-          character.data.spells.spellcasting_ability as ScoreKey
-        ]
-      ) +
-      character.data.proficiency_bonus
+  //   const newList = [...(this.list() ?? []), character]
+  //   console.log(newList)
+
+  //   this.mutate(newList)
+  //   // await CharacterService.addCharacter(character)
+  //   // if there is an error with the api call, we could refetch to make sure the local state is up-to-date with the remote.
+  // }
+
+  // removeCharacter = (slug: string) => {
+  //   this.mutate((this.list() ?? []).filter(character => character.slug !== slug))
+  //   // CharacterService.deleteCharacter(id)
+  //   // if there is an error with the api call, we could refetch to make sure the local state is up-to-date with the remote.
+  // }
+
+  getCharacter = (
+    slug: string
+  ): [
+    CharacterSchema,
+    (updater: (character: CharacterSchema) => void) => void,
+  ] => {
+    // TODO we might want to use createDeepSignal to allow updating only the parts of the store that changes when re-fetching resource: https://github.com/solidjs-community/solid-primitives/tree/main/packages/resource#createdeepsignal
+    const [characterResource, { mutate, refetch }] = createResource(
+      slug,
+      CharacterService.displayCharacter,
+      { initialValue: emptyCharacter }
     )
-  },
 
-  spell_attack_bonus: (character: CharacterSchema) => {
-    return (
-      scoreToSkillModifier(
-        character.data.scores[
-          character.data.spells.spellcasting_ability as ScoreKey
-        ]
-      ) + character.data.proficiency_bonus
+    const [character, setCharacter] = createStore<CharacterSchema>(
+      characterResource()
     )
-  },
 
-  // Recompute the skill modifiers when a characteristic changes
-  ...Object.fromEntries(
-    [
-      ['acrobatics', 'dexterity'],
-      ['animal_handling', 'wisdom'],
-      ['arcana', 'intelligence'],
-      ['athletics', 'strength'],
-      ['deception', 'dexterity'],
-      ['history', 'intelligence'],
-      ['insight', 'wisdom'],
-      ['intimidation', 'charisma'],
-      ['investigation', 'intelligence'],
-      ['medicine', 'wisdom'],
-      ['nature', 'intelligence'],
-      ['perception', 'wisdom'],
-      ['performance', 'charisma'],
-      ['persuasion', 'charisma'],
-      ['religion', 'intelligence'],
-      ['sleight_of_hand', 'dexterity'],
-      ['stealth', 'dexterity'],
-      ['survival', 'wisdom'],
-    ].map(([attribute, secondary]) => [
-      attribute,
-      (character: CharacterSchema) =>
-        scoreToProficiencyModifier(
-          character.data.scores[secondary as ScoreKey],
-          character.data.proficiencies.skills[attribute as SkillProficiencyKey],
-          character.data.proficiency_bonus
+    const abilityKeys = Object.keys(character.data.proficiencies.saves)
+
+    const baseEffects = [
+      // proficiency bonus effect
+      {
+        name: 'base',
+        priority: 10,
```

Comment:
Could these be default values, as we seem to repeat them all the time?

Reply:
I'd be wary of where to set these default, but putting them in `computeEffect` seems good enough.
To be honest, I've over priorities, but feel free to implement it 🙂